allennlp/data/semparse/worlds/nlvr_world.py

```diff
@@ -227,9 +227,17 @@ def _get_number_productions(sentence: str) -> List[str]:
         number_strings = {"one": "1", "two": "2", "three": "3", "four": "4", "five": "5", "six":
                           "6", "seven": "7", "eight": "8", "nine": "9", "ten": "10"}
         number_productions = []
-        for word, numeral in number_strings.items():
-            if word in sentence or numeral in sentence:
-                number_productions.append(f"e -> {numeral}")
+        tokens = sentence.split()
+        for token in tokens:
+            if token.isdigit():
```

Comment:
It'd be simpler to just have `digits = number_strings.values()` above, and checking for existence in that set.

Reply:
Done.